browser/ui/browser_commands.cc

```diff
@@ -845,4 +847,58 @@ void ToggleAllBookmarksButtonVisibility(Browser* browser) {
       !prefs->GetBoolean(brave::bookmarks::prefs::kShowAllBookmarksButton));
 }
 
+bool CanOpenNewSplitViewForActiveTab(Browser* browser) {
+  CHECK(base::FeatureList::IsEnabled(tabs::features::kBraveSplitView));
+  if (!browser->is_type_normal()) {
+    return false;
+  }
+
+  auto* model = browser->tab_strip_model();
+  auto* active_contents = model->GetActiveWebContents();
+  if (!active_contents) {
+    return false;
+  }
+
+  const int active_tab_index = model->GetIndexOfWebContents(active_contents);
+  auto* split_view_data = SplitViewBrowserData::FromBrowser(browser);
+  if (!split_view_data) {
+    //  can happen on startup
+    return false;
+  }
+
+  return !split_view_data->IsTabTiled(model->GetTabHandleAt(active_tab_index));
+}
+
+void NewSplitViewForActiveTab(Browser* browser) {
+  CHECK(base::FeatureList::IsEnabled(tabs::features::kBraveSplitView));
+  if (!CanOpenNewSplitViewForActiveTab(browser)) {
+    return;
+  }
+
+  auto* model = browser->tab_strip_model();
+  int active_tab_index =
+      model->GetIndexOfWebContents(model->GetActiveWebContents());
+  auto* split_view_data = SplitViewBrowserData::FromBrowser(browser);
+  chrome::AddTabAt(browser, GURL("chrome://newtab"), active_tab_index + 1,
+                   /*foreground*/ true);
```

Comment:
Does this mean focus will be transferred into the `split` tab? I guess that's probably a good thing :smile:

Reply:
That's true. |foreground| indicates that the new tab will be the active web contents.